### PR TITLE
Tutorial Review for Tethys 4.3

### PR DESCRIPTION
**Pull request by swainn**
### Description
This pull request include updates to all of the tutorials after a complete review. This update was necessary because of the changes to the app scaffold and other changes in Tethys 4.3. It also includes a few code changes that address issues discovered while reviewing the tutorials.

### Changes Made to Code:
 - All tutorials were reviewed and updated
 - Key Concepts tutorial was refactored to use the Map Layout instead of the Map View Gizmo for the primary map view. It still illustrates how to use the Map View Gizmo in forms.
 - Added screenshots to the Key Concepts tutorial to match the style of the newer tutorials.
 - Fixed a CSS issue in the app base template that prevented the Map View Gizmo from displaying when height was set to 100%.
 - Linted the default app scaffold template.
 - Fixed an issue with the map tabs content area on the Map Layout that was causing the page to have a scroll bar.

### Quality Checks
 - [] New code is 100% tested
 - [] Code has been formatted
 - [] Code has been linted
 - [] Docstrings for new methods have been added

### CODE REVIEW

**Comment by sdc50** [comment]
@swainn how do you access the docs build for a PR again?

**Comment by sdc50** [comment]
I'm confused about what is happening with the quotas tutorial. It looks like changes where made to the `quotas-solution` branch, that remove all of the code that the tutorial says to add.

**Comment by swainn** [comment]
> @swainn how do you access the docs build for a PR again?

It's the details link next to the docs/readthedocs.org:tethys-platform check (near the bottom).

File: docs/tutorials/google_earth_engine/part_2/clip_by_asset.rst [hunk 1]
```diff
@@ -1,646 +1,648 @@
...
+            }
+
+            # Create ee.Feature from geojson (this is the Upload, b/c ee.Feature is a server object)
+            features.append(ee.Feature(geojson_feature))
+
+        feature_collection = ee.FeatureCollection(features)
+
+        # Get unique folder for each user to story boundary asset
+        user_boundary_asset_path = get_user_boundary_path(user)
+        print(user_boundary_asset_path)
```
Comment (swainn) [inline docs/tutorials/google_earth_engine/part_2/clip_by_asset.rst:307]: Remove this print statement unless you intended for it to be part of the tutorial.

File: docs/tutorials/google_earth_engine/part_2/file_upload.rst [hunk 2]
```diff
@@ -427,6 +427,8 @@ In this step you will add the logic to validate that the file contained in the Z
           - earthengine-api
           - oauth2client
           - geojson
+          - simplejson
```
Comment (swainn) [inline docs/tutorials/google_earth_engine/part_2/file_upload.rst:430]: Is `simplejson` used? Try to stick with the built-in `json` module.

File: docs/tutorials/google_earth_engine/part_2/rest_api.rst [hunk 3]
```diff
@@ -64,7 +64,9 @@ Add ``djangorestframework`` to the ``install.yml`` file to ensure it is installe
             - earthengine-api
             - oauth2client
             - geojson
+            - pandas
             - pyshp
+            - simplejson
```
Comment (swainn) [inline docs/tutorials/google_earth_engine/part_2/rest_api.rst:69]: Is `simplejson` required here?

File: docs/tutorials/google_earth_engine/part_3/prepare.rst [hunk 4]
```diff
@@ -96,7 +96,28 @@ Although using the :file:`gee/params.py` file to store our service account crede
 
     from . import params as gee_account
 
-4. **Delete** :file:`gee/params.py`.
```
Comment (swainn) [inline docs/tutorials/google_earth_engine/part_3/prepare.rst:99]: Seems like this file still needs to be deleted.

File: docs/tutorials/thredds/additional_exercises.rst [hunk 5]
```diff
@@ -1,67 +1,69 @@
...
+The current implementation uses hard-coded values for the color scale range of the style applied to the layer. See ``update_layer`` method of :file:`public/js/leaflet_map.js`:
+
+.. code-block:: javascript
+
+    update_layer = function() {
+        ...
+        var proxyWMSURL = `getWMSImageFromServer?main_url=${encodeURIComponent(m_curr_wms_url)}`;
+
+        // Layer
+        m_layer = L.tileLayer.wms(proxyWMSURL, {
```
Comment (swainn) [inline docs/tutorials/thredds/additional_exercises.rst:37]: Why the change to the proxy URL?

File: docs/tutorials/websockets.rst [hunk 6]
```diff
@@ -28,7 +29,43 @@ If you wish to use the advanced solution as a starting point:
...
+1. Begin by adding a new file called ``home.html`` to the ``templates/dam_inventory`` directory. Add the following code to the file:
+
+    .. code-block:: html+django
+
+        {% extends "tethys_layouts/map_layout/map_layout.html" %}
+        {% load tethys_gizmos %}
+
+
+    This code extends the default map layout template, while allowing us to make manual additions later. It also prepares the template for including 
+    custom Tethys Gizmos later.
```
Comment (swainn) [inline docs/tutorials/websockets.rst:45]: Remove the newline between these two lines since they are the same paragraph. Only use newlines between paragraphs. Use the word wrap feature of your editor to wrap the text while you are editing.

File: docs/tutorials/dask/new_app_project.rst
Comment (swainn) [inline docs/tutorials/dask/new_app_project.rst:1]: For some reason, every line of this file (and many others) is showing up as changed. This is usually due to white space changes. Can you make sure that your editor is set to use LF (linux) new lines (not CRLF, windows). If you suspect a lot of the files were edited with CRLF, use the `dos2unix` program on the docs dir to change them all to LF line endings.

**Review by swainn — COMMENTED** [review]
See comments.

File: docs/tutorials/google_earth_engine/part_2/file_upload.rst (hunk 2 above)
Comment (jakeymac) [reply docs/tutorials/google_earth_engine/part_2/file_upload.rst:430]: it's used starting in the REST API step for handling json errors.

File: docs/tutorials/google_earth_engine/part_3/prepare.rst (hunk 4 above)
Comment (jakeymac) [reply docs/tutorials/google_earth_engine/part_3/prepare.rst:99]: This step was moved to step 5, it's still removed.

File: docs/tutorials/thredds/additional_exercises.rst (hunk 5 above)
Comment (jakeymac) [reply docs/tutorials/thredds/additional_exercises.rst:37]: It's needed to avoid a CORS issue with the endpoint

File: docs/tutorials/dask/new_app_project.rst
Comment (jakeymac) [reply docs/tutorials/dask/new_app_project.rst:1]: done

**Review by swainn — COMMENTED** [review]
I'm not allowed officially approve, since I have commits in this PR, but I approve.

**Comment by swainn** [comment]
> I'm not allowed officially approve, since I have commits in this PR, but I approve.

I rescind my approval. I pulled up the check build of the docs and noticed that the logo looks terrible:

image

Remove the "Tethys Platform" text and use some custom CSS to make the logo bigger.

**Comment by swainn** [comment]
> I'm confused about what is happening with the quotas tutorial. It looks like changes where made to the `quotas-solution` branch, that remove all of the code that the tutorial says to add.

@jakeymac please look into this.

**Comment by jakeymac** [comment]
> I'm confused about what is happening with the quotas tutorial. It looks like changes where made to the `quotas-solution` branch, that remove all of the code that the tutorial says to add.

@sdc50 I've updated the tutorial.

**Comment by swainn** [comment]
> > I'm not allowed officially approve, since I have commits in this PR, but I approve.
> 
> I rescind my approval. I pulled up the check build of the docs and noticed that the logo looks terrible:
> 
> image
> 
> Remove the "Tethys Platform" text and use some custom CSS to make the logo bigger.

I rescind my previous rescinding. I now approve this PR.
